launch/client.py

```diff
@@ -902,6 +902,7 @@ def _sync_request(
         resp = self.connection.post(
             payload=payload,
             route=f"{SYNC_TASK_PATH}/{endpoint_id}",
+            handle_bad_response=False,
```

Comment:
if we return http 200 + "failed" as part of the sync endpoint failing response, we can get rid of this "handle_bad_response" thing